rest/src/main/java/org/jboss/pnc/rest/provider/BuildRecordProvider.java

```diff
@@ -660,63 +696,76 @@ public BuildRecordRest getLatestBuildRecord(Integer configId, boolean executedOn
      * complete. This abstraction allows clients to query for a list of all builds whether running or completed.
      */
     private CollectionInfo<BuildRecordRest> getBuilds(Integer pageIndex, Integer pageSize, String sort, String orFindByBuildConfigurationName, String andFindByBuildConfigurationName, String... rsqlQueries) {
+
+        logger.debug(
+                "Get running and completed build records by pageIndex: {}, pageSize: {}, sort: {}, orFindByBuildConfigurationName: {}, andFindByBuildConfigurationName: {}, rsqlQueries: {}",
+                pageIndex, pageSize, sort, orFindByBuildConfigurationName, andFindByBuildConfigurationName, rsqlQueries);
+
+        // Combine predicates of both search query and the additional filters (e.g.userId, buildConfigurationId)
         List<Predicate<BuildRecord>> dbAndPredicatesList = Arrays.stream(rsqlQueries)
                 .filter(rsqlString -> rsqlString != null && !rsqlString.isEmpty())
                 .map(p -> rsqlPredicateProducer.getPredicate(BuildRecord.class, p))
                 .collect(Collectors.toList());
         List<Predicate<BuildRecord>> dbOrPredicateList = new ArrayList<>();
 
+        // Combine rsqlQueries AND-ing all the single queries
         String combinedQueries = combineRsqlQueriesMatchAll(rsqlQueries);
-        if (!StringUtils.isEmpty(orFindByBuildConfigurationName)) {
-            //add steam condition
+
+        // An AND condition (andFindByBuildConfigurationName) gets higher precedence over an OR condition
```

Comment:
If and AND condition (andFindByBuildConfigurationName) is specified, it gets processed. Else if an OR condition (orFindByBuildConfigurationName) is specified, it gets processed. If BOTH are specified (non sensible), make sure the only AND condition is picked.